File: lib/core_services/comic/mixin/comic_follow_mixin.dart
import 'package:hoyomi/core_services/comic/main.dart';
import 'package:hoyomi/core_services/mixin/auth_mixin.dart';

mixin ComicFollowMixin on AuthMixin {
  Future<Paginate<ComicFollow>> getFollows({required int page});

  Future<bool> isFollow({required String comicId});
  Future<void> setFollow({
    required String comicId,
    required MetaComic metaComic,
    required bool value,
  });
}

File: lib/core_services/comic/mixin/comic_follow_mixin.ts
import type { Paginate } from "../../interfaces/paginate"
import type { MetaComic } from "../interfaces/meta_comic"
import type { ComicFollow } from "../interfaces/comic_follow"
import { ComicAuthMixin } from "./comic_auth_mixin"

export abstract class ComicFollowMixin extends ComicAuthMixin {
  abstract getFollows(page: number): Promise<Paginate<ComicFollow>>

  abstract isFollow(comicId: string): Promise<boolean>

  abstract setFollow(params: {
    comicId: string
    metaComic: MetaComic
    value: boolean
  }): Promise<void>
}
